records_mover/records/validated_records_hints.py

```diff
@@ -10,142 +12,175 @@
+                                        'compression',
+                                        default=None)
+    quoting = Hint[HintQuoting](HintQuoting,  # type: ignore
+                                'quoting',
+                                default='minimal')
+    escape = Hint[HintEscape](HintEscape,  # type: ignore
+                              'escape',
+                              default='\\')
+    encoding = Hint[HintEncoding](HintEncoding,  # type: ignore
+                                  'encoding',
+                                  default='UTF8')
+    dateformat = Hint[HintDateFormat](HintDateFormat,  # type: ignore
+                                      'dateformat',
+                                      default='YYYY-MM-DD')
+    timeonlyformat = Hint[HintTimeOnlyFormat](HintTimeOnlyFormat,  # type: ignore
+                                              'timeonlyformat',
+                                              default="HH24:MI:SS")
+
+
+HintA = TypeVar('HintA')
+
+HintB = TypeVar('HintB')
+
+
+class HintValidator:
+    def __init__(self,
+                 fail_if_cant_handle_hint: bool,
+                 hints: RecordsHints):
+        self.fail_if_cant_handle_hint = fail_if_cant_handle_hint
+        self.hints = hints
+
+    def validate_literal(self, hint: Hint[HintA]) -> HintA:
+        # MyPy doesn't like looking for a generic optional string
+        # in a list of specific optional strings.  It's wrong;
+        # that's perfectly safe
+        x: object = self.hints[hint.hint_name]
+        try:
+            i = hint.valid_values.index(x)  # type: ignore
+            return hint.valid_values[i]
+        except ValueError:
+            cant_handle_hint(fail_if_cant_handle_hint=self.fail_if_cant_handle_hint,
+                             hint_name=hint.hint_name,
+                             hints=self.hints)
+            return hint.default
+
+    # TODO: Can this method be typesafe?  HintT doesn't even mean anything here
+    def validate_hint(self, hint: Hint[HintB]) -> HintB:
+        assert is_literal_type(hint.type_)
+
+        return self.validate_literal(hint)
+
+    def validate(self) -> 'ValidatedRecordsHints':
         def validate_boolean(hint_name: str) -> Union[Literal[True], Literal[False]]:
-            x = hints[hint_name]
+            x = self.hints[hint_name]
             if x is True:
                 return True
             if x is False:
                 return False
-            cant_handle_hint(fail_if_cant_handle_hint=fail_if_cant_handle_hint,
+            cant_handle_hint(fail_if_cant_handle_hint=self.fail_if_cant_handle_hint,
                              hint_name=hint_name,
-                             hints=hints)
+                             hints=self.hints)
             return True
 
         def validate_string(hint_name: str) -> str:
-            x = hints[hint_name]
+            x = self.hints[hint_name]
             if isinstance(x, str):
                 return x
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker